src/engine/engine.rs

```diff
@@ -84,23 +75,39 @@ impl From<nix::Error> for EngineError {
 }
 
 pub trait Pool: Debug {
-    fn create_filesystem(&mut self,
-                         name: &str,
-                         mount_point: &str,
-                         quota_size: Option<u64>)
-                         -> EngineResult<()>;
-    fn create_snapshot(&mut self, snapshot_name: &str, source: &str) -> EngineResult<()>;
+    /// Creates the filesystems specified by specs.
+    /// Returns a list of the names of filesystems actually created.
+    /// Returns an error if any filesystems of the given name occur.
+    fn create_filesystems<'a, 'b, 'c>(&'a mut self,
+                                      mut specs: Vec<(&'b str, &'c str, Option<u64>)>)
```

Comment:
do we need mut in `mut specs` here, or only in the implementation? We're moving ownership, of course it's mut. (really a question about Rust I don't know the answer to)